frontend/packages/console-shared/src/components/utils/with-resources.tsx

```diff
@@ -0,0 +1,167 @@
+import * as React from 'react';
+import * as PropTypes from 'prop-types';
+import * as _ from 'lodash-es';
+import { connect } from 'react-redux';
+import { Map as ImmutableMap } from 'immutable';
+
+import { Firehose, inject } from '@console/internal/components/utils';
+
+const checkErrors = (resources, onError) => {
+  if (resources.length > 0) {
+    if (onError) {
+      onError(resources);
+    }
+    resources.forEach(resource => {
+      const errorMessage = _.get(
+        resource.error,
+        'json.message',
+        `Error occured while loading ${resource.resourceConfig.resource.kind}`
+      );
+      const errorCode = _.get(resource.error, 'json.code', '');
+      const error = errorCode ? `${errorCode}: ${errorMessage}` : errorMessage;
+      console.warn(error); // eslint-disable-line
+    });
+  }
+};
+
+/*
+ * Firehose helper
```

Comment:
Can you explain what this does and why it's needed?

Reply:
This seems to be a KubeVirt utility, ping @rawagner @suomiy

https://github.com/kubevirt/web-ui/blob/master/frontend/public/kubevirt/components/utils/withResources.jsx

Reply:
Yes, it is KubeVirt utility which simplifies work with Firehose and adds some features.
For example:
- `Firehose` wont render children if model does not exist in k8s. `WithResources` will - both cases has their place
- `Resources` accepts resourceMap which is object instend of array as in Firehose which brings us a few enhancements:

  - you need to define key (which is used as firehose resource prop - just makes it easier if you have more queries for the same model, object forces you to use unique keys)

  - you can define if resource from resourceMap is required - if yes we wait with child rendering until the resource is loaded (or we can show Loading)
- there is also way to define your own error handling
- you no longer need to check firehose's loaded prop - when injected resource is undefined it means it is still loading